### PR TITLE
Fix prime array length

### PR DESCRIPTION
**Pull request by sleeepyjack**

### CODE REVIEW

File: include/cuco/detail/prime.hpp [hunk 1]
```diff
@@ -25,7 +25,7 @@
 namespace cuco {
 namespace detail {
 
-inline constexpr std::array<int64_t, 140746> primes = {
+inline constexpr std::array<int64_t, 140739> primes = {
```
Comment (PointKernel) [inline include/cuco/detail/prime.hpp:28]:
```suggestion
inline constexpr std::array<uint64_t, 140739> primes = {
```
I recall you changed the data type to `uint64_t` in `refactor` branch.

Comment (sleeepyjack) [reply include/cuco/detail/prime.hpp:28]: Yes, the primary reason for the change was that I needed this for the new `next_prime` function in the `refactor` branch to be able to support the case where `cuco::extent::size_type == size_t`. I does not affect the `dev` branch so I thought we leave it for now.

**Review by PointKernel — APPROVED** [review]

**Review by karthikeyann — APPROVED** [review]

**Comment by jrhemstad** [comment]
Funny, I was originally going to suggest on the other PR to use CTAD to avoid the possibility of mistaking the length but I didn't want to be pedantic :) 

The slightly annoying thing is you'd have to append the `ll` suffix to every value to make them all the same type to make CTAD work: https://godbolt.org/z/jW9PETvKh

**Comment by sleeepyjack** [comment]
> Funny, I was originally going to suggest on the other PR to use CTAD to avoid the possibility of mistaking the length but I didn't want to be pedantic :)
> 
> The slightly annoying thing is you'd have to append the `ll` suffix to every value to make them all the same type to make CTAD work: https://godbolt.org/z/jW9PETvKh

Good idea! Adressed in [9ac7d40](https://github.com/NVIDIA/cuCollections/pull/255/commits/9ac7d405995f09ec8c340ec887cc827af2dc231e). The diff is of course huge but it looks like this:
```cpp
inline constexpr auto primes = std::array{
  2ull,           3ull,           5ull,           7ull,           13ull,          19ull,
```
,i.e., I also switched to `uint64_t` so I don't need to touch this commit again when it is merged into the `rebase` branch.

**Comment by sleeepyjack** [comment]
CUDA 11.0 is showing a segfault during the build step. Is this a compiler issue?
This reminds me that we wanted to switch to CUDA 11.5 as the minimum required CTK version. (see #251)

```
[ 13%] Linking CXX executable DYNAMIC_MAP_TEST
[ 13%] Built target DYNAMIC_MAP_TEST
[ 14%] Building CUDA object tests/CMakeFiles/STATIC_MULTIMAP_TEST.dir/static_multimap/custom_pair_retrieve_test.cu.o
Segmentation fault (core dumped)
make[2]: *** [tests/CMakeFiles/STATIC_MULTIMAP_TEST.dir/build.make:77: tests/CMakeFiles/STATIC_MULTIMAP_TEST.dir/static_multimap/custom_pair_retrieve_test.cu.o] Error 139
make[1]: *** [CMakeFiles/Makefile2:1165: tests/CMakeFiles/STATIC_MULTIMAP_TEST.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
```

**Comment by PointKernel** [comment]
rerun tests

**Comment by PointKernel** [comment]
rerun tests

**Comment by sleeepyjack** [comment]
Shall I revert the last commit for now?

**Comment by PointKernel** [comment]
> Shall I revert the last commit for now?

Yeah, can you please add a `TODO` comment reminding us of the CTAD improvement once we no longer support 11.0?

**Review by PointKernel — APPROVED** [review]

**Pull request merged**